Makefile.am

```diff
@@ -12,9 +12,8 @@ applications_DATA = htop.desktop
 pixmapdir = $(datadir)/pixmaps
 pixmap_DATA = htop.png
 
-AM_CFLAGS += -pedantic -Wall $(wextra_flag) -std=c99 -D_XOPEN_SOURCE_EXTENDED -DSYSCONFDIR=\"$(sysconfdir)\" -I"$(top_srcdir)/$(my_htop_platform)"
```

Comment:
-(W)pedantic is not included in -Wall, why drop it?

Reply:
The statement expressions, `({ ... })`, inside the macros are not C99 (see https://gcc.gnu.org/onlinedocs/gcc/Statement-Exprs.html).
So `-pedantic` warns about them.

Reply:
Hm, and you think this patch is important enough to drop C99 compatibility for some loops in preprocessor instructions? (honest question)

Reply:
Reworked in a C99 compliant fashion